pm/sendermonitor.go

```diff
@@ -286,3 +270,76 @@ func (sm *senderMonitor) cleanup() {
 		}
 	}
 }
+
+func (sm *senderMonitor) redeemWinningTicket(ticket *SignedTicket) (err error) {
+	maxFloat, e := sm.MaxFloat(ticket.Ticket.Sender)
+	if e != nil {
+		err = e
+		return
+	}
+
+	// if max float is zero, there is no claimable reserve left or reserve is 0
+	if maxFloat.Cmp(big.NewInt(0)) <= 0 {
+		sm.QueueTicket(ticket)
+		err = errors.Errorf("max float is zero")
+		return
+	}
+
+	// If max float is insufficient to cover the ticket face value, queue
+	// the ticket to be retried later
+	if maxFloat.Cmp(ticket.Ticket.FaceValue) < 0 {
+		sm.QueueTicket(ticket)
+		err = fmt.Errorf("insufficient max float sender=%v faceValue=%v maxFloat=%v", ticket.Ticket.Sender.Hex(), ticket.Ticket.FaceValue, maxFloat)
+		return
+	}
+
+	// Subtract the ticket face value from the sender's current max float
+	// This amount will be considered pending until the ticket redemption
+	// transaction confirms on-chain
+	sm.subFloat(ticket.Ticket.Sender, ticket.Ticket.FaceValue)
+
+	defer func() {
+		// Add the ticket face value back to the sender's current max float
+		// This amount is no longer considered pending since the ticket
+		// redemption transaction either confirmed on-chain or was not
+		// submitted at all
+		//
+		// TODO(yondonfu): Should ultimately add back only the amount that
+		// was actually successfully redeemed in order to take into account
+		// the case where the ticket was not redeemd for its full face value
+		// because the reserve was insufficient
+		e := sm.addFloat(ticket.Ticket.Sender, ticket.Ticket.FaceValue)
```

Comment:
IIUC the reason a named return arg is being used here is so we can return the error returned by `addFloat()` instead of just logging it.

The scenario where `addFloat()` returns an error is when the amount passed in is for some reason greater than a sender's pendingAmount. This should never happen, but the check in `addFloat()` is there just to be safe. An alternative is to instead update `addFloat()` to subtract min(pendingAmount, amount) instead of returning an error and then we wouldn't need to used named return args here.

Reply:
I decided to leave it as is for now because we now return an error from both addFloat and subFloat regardless of the math operations as those functions are now responsible for sending the max float update. 

We can always revisit this and make the caller of those (eg `redeemWinningTicket`) responsible for sending updates if necessary.